File: app/src/main/java/ch/threema/app/listeners/ContactSettingsListener.java
package ch.threema.app.listeners;

import androidx.annotation.AnyThread;

public interface ContactSettingsListener {
    @AnyThread
    default void onSortingChanged() {}

    @AnyThread
    default void onNameFormatChanged() {}

    @AnyThread
    default void onAvatarSettingChanged() {}

    @AnyThread
    default void onInactiveContactsSettingChanged() {}

    @AnyThread
    default void onNotificationSettingChanged(String uid) {}
}

File: app/src/main/java/ch/threema/app/listeners/ContactSettingsListener.kt
package ch.threema.app.listeners

import androidx.annotation.AnyThread
import ch.threema.data.datatypes.ContactNameFormat

interface ContactSettingsListener {
    @AnyThread
    fun onSortingChanged() {
    }

    @AnyThread
    fun onNameFormatChanged(nameFormat: ContactNameFormat) {
    }

    @AnyThread
    fun onIsDefaultContactPictureColoredChanged(isColored: Boolean) {
    }

    @AnyThread
    fun onShowContactDefinedAvatarsChanged(shouldShow: Boolean) {
    }

    @AnyThread
    fun onInactiveContactsSettingChanged() {
    }

    @AnyThread
    fun onNotificationSettingChanged(uid: String?) {
    }
}
